frontend/src/CreateAnnotationCampaign.tsx

```diff
@@ -195,102 +176,92 @@ class CreateAnnotationCampaign extends Component<CACProps, CACState> {
-          // Server returned 401 which means token was revoked
-          document.cookie = 'token=;max-age=0;path=/';
-          window.location.reload();
-        }
-        this.setState({
-          error: err
-        });
       }),
       this.getUsers.set('Authorization', 'Bearer ' + this.props.app_token).then(req => {
-        let users = req.body.map(user => { return { id: user.id, name: user.email }; });
+        let users = req.body.map((user: any) => { return { id: user.id, name: user.email }; });
         this.setState({
           annotator_choices: utils.arrayToMap(users, 'id')
         });
-      }).catch(err => {
-        if (err.status && err.status === 401) {
-          // Server returned 401 which means token was revoked
-          document.cookie = 'token=;max-age=0;path=/';
-          window.location.reload();
-        }
-        this.setState({
-          error: err
-        });
-      })
-    ]);
-
+      }),
+    ])
+    .then(() => this.setState({ error: undefined }))
+    .catch(err => {
+      if (err.status && err.status === 401) {
+        // Server returned 401 which means token was revoked
+        document.cookie = 'token=;max-age=0;path=/';
+        window.location.reload();
+      }
+      this.setState({
+        error: err
+      });
+    });
   }
 
   componentWillUnmount() {
-    this.getDatasets.abort();
-    this.getAnnotationSets.abort();
-    this.getUsers.abort();
-    this.postAnnotationCampaign.abort();
+    if (this.getDatasets) {
+      this.getDatasets.abort();
+    }
+    if (this.getAnnotationSets) {
+      this.getAnnotationSets.abort();
+    }
+    if (this.getUsers) {
+      this.getUsers.abort();
+    }
+    if (this.getConfidenceSets) {
+      this.getConfidenceSets.abort();
+    }
+    if (this.postAnnotationCampaign) {
+      this.postAnnotationCampaign.abort();
+    }
   }
 
-  handleNameChange = (event: SyntheticEvent<HTMLInputElement>) => {
+  handleNameChange = (event: ChangeEvent<HTMLInputElement>) => {
     this.setState({new_ac_name: event.currentTarget.value});
   }
 
-  handleDescChange = (event: SyntheticEvent<HTMLInputElement>) => {
+  handleDescChange = (event: ChangeEvent<HTMLTextAreaElement>) => {
     this.setState({new_ac_desc: event.currentTarget.value});
   }
 
-  handleDatasetChanged = (event: SyntheticInputEvent<HTMLSelectElement>) => {
-    let new_ac_dataset = {};
-    let spectro_choices = {};
+  handleDatasetChanged = (event: ChangeEvent<HTMLSelectElement>) => {
+    let new_ac_dataset: any; // TODO HERE
+    let spectro_choices: choices_type = new Map();
     if (event.target.value !== '') {
       let dataset_id = parseInt(event.target.value, 10);
-      new_ac_dataset = this.state.dataset_choices.get(dataset_id);
-      spectro_choices = utils.arrayToMap(new_ac_dataset.spectros, 'id');
+      new_ac_dataset = this.state.dataset_choices.get(dataset_id)!;
+      spectro_choices = utils.arrayToMap(new_ac_dataset.spectros, 'id'); // TODO HERE
```

Comment:
What is has to be done here ? Is it a forgotten dev comment ?

Reply:
As `new_ac_datasets` is hard to type because it is inconstently used through the file, I added todos at each occurence.
I made the comments clearer in last commit.